### PR TITLE
[Back-port]  Removing btrfs specific tests from py2-lts branch from LVM script

### PR DESCRIPTION
**Pull request by manvanthar**
Removing btrfs specific tests from py2-lts branch from LVM script

Signed-off-by: Manvanthara B Puttashankar <manvanth@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by manvanthar** [comment]
Command line: /usr/bin/avocado run lvsetup.py -m lvsetup.py.data/lvsetup.yaml --show-job-log

Avocado version: 69.0

Config files read (in order):
/usr/lib/python2.7/site-packages/avocado_framework-69.0-py2.7.egg/avocado/etc/avocado/avocado.conf
/usr/lib/python2.7/site-packages/avocado_framework-69.0-py2.7.egg/avocado/etc/avocado/conf.d/gdb.conf
/usr/lib/python2.7/site-packages/avocado_framework-69.0-py2.7.egg/avocado/etc/avocado/conf.d/jobscripts.conf
/usr/lib/python2.7/site-packages/avocado_framework-69.0-py2.7.egg/avocado/etc/avocado/conf.d/result_upload.conf
/usr/lib/python2.7/site-packages/avocado_framework-69.0-py2.7.egg/avocado/etc/avocado/conf.d/resultsdb.conf
/etc/avocado/avocado.conf
/etc/avocado/conf.d/gdb.conf
/etc/avocado/conf.d/jobscripts.conf
/root/.config/avocado/avocado.conf

Avocado config:
Section.Key                              Value
datadir.paths.base_dir                   /root/tests
datadir.paths.test_dir                   /root/tests/tests
datadir.paths.data_dir                   /root/tests/data
datadir.paths.logs_dir                   /root/tests/results
sysinfo.collect.enabled                  True
sysinfo.collect.commands_timeout         -1
sysinfo.collect.installed_packages       False
sysinfo.collect.profiler                 True
sysinfo.collect.locale                   C
sysinfo.collect.per_test                 True
sysinfo.collectibles.commands            /etc/avocado/sysinfo/commands
sysinfo.collectibles.files               /etc/avocado/sysinfo/files
sysinfo.collectibles.profilers           /etc/avocado/sysinfo/profilers
runner.output.colored                    True
runner.output.utf8                       
remoter.behavior.reject_unknown_hosts    False
remoter.behavior.disable_known_hosts     False
job.output.loglevel                      debug
restclient.connection.hostname           localhost
restclient.connection.port               9405
restclient.connection.username           
restclient.connection.password           
plugins.disable                          []
plugins.skip_broken_plugin_notification  []
plugins.loaders                          ['file', '@DEFAULT']
gdb.paths.gdb                            /usr/bin/gdb
gdb.paths.gdbserver                      /usr/bin/gdbserver
plugins.jobscripts.pre                   /etc/avocado/scripts/job/pre.d/
plugins.jobscripts.post                  /etc/avocado/scripts/job/post.d/
plugins.jobscripts.warn_non_existing_dir False
plugins.jobscripts.warn_non_zero_status  True

Avocado Data Directories:

base     /root/tests
tests    /root/tests/tests
data     /root/tests/data
logs     /root/tests/results/job-2019-10-09T05.55-83f599b

Multiplex tree representation:
 \-- run
      \-- filesystem
           #== btrfs
           #== ext2
           #== ext3
           #== ext4
           #== xfs

Multiplex variants (5):
Variant btrfs-dece:    /run/filesystem/btrfs
Variant ext2-c842:    /run/filesystem/ext2
Variant ext3-56e8:    /run/filesystem/ext3
Variant ext4-6fb3:    /run/filesystem/ext4
Variant xfs-292c:    /run/filesystem/xfs
Temporary dir: /var/tmp/avocado_ae2Z59/avocado_job_5u_N9g

Job ID: 83f599bb46275108bdd36e88527581d2f1bbca10

Not logging /sys/devices/system/cpu/cpu0/cpufreq/scaling_governor (file does not exist)
Not logging /proc/pci (file does not exist)
Commands configured by file: /etc/avocado/sysinfo/commands
Files configured by file: /etc/avocado/sysinfo/files
Profilers configured by file: /etc/avocado/sysinfo/profilers
INIT 01-lvsetup.py:Lvsetup.test;btrfs-dece
PARAMS (key=timeout, path=*, default=None) => None
Test metadata:
  filename: /root/avocado-misc-tests/io/disk/lvsetup.py
  teststmpdir: /var/tmp/avocado_4xoKnv
  workdir: /var/tmp/avocado_ae2Z59/avocado_job_5u_N9g/01-lvsetup.py_Lvsetup.test_btrfs-dece
START 01-lvsetup.py:Lvsetup.test;btrfs-dece
DATA (filename=output.expected) => NOT FOUND (data sources: variant, test, file)
Not logging /sys/devices/system/cpu/cpu0/cpufreq/scaling_governor (file does not exist)
Not logging /proc/pci (file does not exist)
PARAMS (key=disks, path=*, default=None) => '/dev/sdb'
PARAMS (key=vg_name, path=*, default=avocado_vg) => 'avocado_vg'
PARAMS (key=lv_name, path=*, default=avocado_lv) => 'avocado_lv'
PARAMS (key=fs, path=*, default=ext4) => 'btrfs'

Reproduced traceback from: /usr/lib/python2.7/site-packages/avocado_framework-69.0-py2.7.egg/avocado/core/test.py:831
Traceback (most recent call last):
  File "/root/avocado-misc-tests/io/disk/lvsetup.py", line 63, in setUp
    self.cancel("btrfs is not supported with rhel")
  File "/usr/lib/python2.7/site-packages/avocado_framework-69.0-py2.7.egg/avocado/core/test.py", line 1057, in cancel
    raise exceptions.TestCancel(message)
TestCancel: btrfs is not supported with RHEL 7.4 onwards

CANCEL 01-lvsetup.py:Lvsetup.test;btrfs-dece -> TestCancel: btrfs is not supported with RHEL 7.4 onwards

File: io/disk/lvsetup.py [hunk 1]
```diff
@@ -57,6 +57,10 @@ def setUp(self):
         if self.fs_name == 'xfs':
             pkg = 'xfsprogs'
         if self.fs_name == 'btrfs':
+            if distro.detect().name == "rhel" and \
+               int(distro.detect().version) >= 7 and \
+               int(distro.detect().release) >= 4 :
```
Comment (PraveenPenguin) [inline io/disk/lvsetup.py]: ah it has the false result for RHEL8 or RHEL8.1 

can below snippt code will work 
 ```
if distro.detect().name == "rhel"  : 
    if int(distro.detect().version) >= 7 :
         self.cancel("btrfs is not supported with RHEL 7.4 onwards")
   if int(distro.detect().version) == 7 and   int(distro.detect().release) >= 4 :
         self.cancel("btrfs is not supported with RHEL 7.4 onwards")
````

Comment (manvanthar) [reply io/disk/lvsetup.py]: can be done praveen.   But tricky part is,  few RHEL8 system works (like as in narasimhans system it gave correct values) where as my system and as well yours have weird behavior with RHEl8.0.

Changes can be done thanks!!

@narasimhan-v

**Comment by manvanthar** [comment]
Fix in PR##1345

**Pull request closed without merging**